### DISCUSSION TITLE
How to write the MetaFunction with defer data?

### DISCUSSION THREAD

**Discussion by zipme** [remix-run/remix#5559] [Q&A]
```ts
export const meta: MetaFunction<typeof loader> = async ({ data }) => {
  // assuming we have a deferredData which is still a promise
  const deferredData =  data.someDeferData 
};
```
Is there a way we can return a default meta data and swap it with the final one when the data is resolved? 

To give an example, I have a search page on my website, and I want the default browser title to show "Search Result | Website". However, once the search result is resolved, I want it to be displayed as "(203) Search Result | Website".

**Comment by sergiodxa** [comment]
If a piece of data is required inside MetaFunction, then that data can't be deferred.

For your use case in particular, you could use an effect to replace the document.title once the data is resolved probably.
